### PR TITLE
[5.0] Make the includes check easier to read

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Makes the check if a configuration exists in the include files easier to read as it returns early when a condition doesn't exist.

### Testing Instructions
- Remove configuration.php
- Go to the root page of the joomla installation
- Remove installation folder
- Go to the root page of the joomla installation

### Actual result BEFORE applying this Pull Request
- The installation screen is shown
- "No configuration file found and no installation code available. Exiting..." message is shown

### Expected result AFTER applying this Pull Request
- The installation screen is shown
- "No configuration file found and no installation code available. Exiting..." message is shown

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/includes/framework.php [hunk 1]
```diff
@@ -15,25 +15,27 @@
...
         exit;
     }
+
+    if (JPATH_ROOT === JPATH_PUBLIC) {
+        header('Location: ' . substr($_SERVER['REQUEST_URI'], 0, strpos($_SERVER['REQUEST_URI'], 'index.php')) . 'installation/index.php');
+
+        exit;
+    }
+
+    echo 'Installation from a public folder is not supported, revert your Server configuration to point at the Joomla\'s root folder to continue.';
```
Comment (brianteeman) [inline administrator/includes/framework.php]:
```suggestion
    echo 'Installation from a public folder is not supported, revert your Server configuration to point at Joomla\'s root folder to continue.';
```

File: includes/framework.php [hunk 2]
```diff
@@ -15,25 +15,27 @@
...
-        echo 'No configuration file found and no installation code available. Exiting...';
+    }
+
+    if (JPATH_ROOT === JPATH_PUBLIC) {
+        header('Location: ' . substr($_SERVER['REQUEST_URI'], 0, strpos($_SERVER['REQUEST_URI'], 'index.php')) . 'installation/index.php');
 
         exit;
     }
+
+    echo 'Installation from a public folder is not supported, revert your Server configuration to point at the Joomla\'s root folder to continue.';
```
Comment (brianteeman) [inline includes/framework.php]:
```suggestion
    echo 'Installation from a public folder is not supported, revert your Server configuration to point at Joomla\'s root folder to continue.';
```

**Comment by sandewt** [comment]
Proposal: If you change the following code, both files are identical except for line 4  (administator / site).

Change this code

https://github.com/joomla/joomla-cms/blob/e6a7e951f1a08184443fd6862ef8cdc2548b0a17/administrator/includes/framework.php#L81

to that of

https://github.com/joomla/joomla-cms/blob/e6a7e951f1a08184443fd6862ef8cdc2548b0a17/includes/framework.php#L81

**Comment by laoneo** [comment]
I guess this is done on purpose as  sites can include the index.php in an own script and then define that variable. If you want to change this, then please in another pr, but I'm not sure about the consequences, perhaps @wilsonge can shed some light into this.

**Comment by sandewt** [comment]
I have tested this item :red_circle: unsuccessfully on b129c8d9f7a5352112047f48351de93932acd29e<br /><br />Remove configuration.php - done
Go to the root page of the joomla installation - done
Error message appears: The requested URL was not found on this server.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41053">issues.joomla.org/tracker/joomla-cms/41053</a>.</sub>

**Comment by laoneo** [comment]
Can you post a screenshot of the error message and the url you have opened and the url you get redirected?

**Comment by sandewt** [comment]
issue-test

**Comment by laoneo** [comment]
Can you try with the latest fix?

**Comment by wilsonge** [comment]
> I guess this is done on purpose as sites can include the index.php in an own script and then define that variable. If you want to change this, then please in another pr, but I'm not sure about the consequences, perhaps @wilsonge can shed some light into this.

I can't shed any proper light. I mean it's always been intended that you can override defines.php rather than index.php at the top level of each app (https://github.com/joomla/joomla-cms/blob/4.3-dev/includes/app.php#L16-L23) so I assume this allows setting JDEBUG there. No clue why you'd choose to set it there. But making it consistent also makes sense to me. 

API should be included in this PR as it's based on administrator https://github.com/joomla/joomla-cms/blob/4.3-dev/api/includes/framework.php

**Comment by laoneo** [comment]
Should then API get the same logic as admin and site got in #40509? If yes, can you comment on the original pr? I will then update this one when the rest is done.

**Comment by sandewt** [comment]
> Can you try with the latest fix?

Works, expected result.

**Comment by laoneo** [comment]
Can you mark it as a successful test?

**Comment by wilsonge** [comment]
> Should then API get the same logic as admin and site got in https://github.com/joomla/joomla-cms/pull/40509? If yes, can you comment on the original pr? I will then update this one when the rest is done.

Ahh we error out instead of redirecting - one day I'll read code correctly the first time. Yeah just ignore me and I'll get back into my box

**Comment by sandewt** [comment]
> Can you mark it as a successful test?

Why these two files? Only file 1 seems to be addressed in this test. I access the root page of the joomla installation through the administrator (administrator/index.php?option=com_joomlaupdate). Or am I not testing it correctly?

C:\xampp\htdocs\bugtesting5\joomla\administrator\includes\framework.php (file 1)

C:\xampp\htdocs\bugtesting5\joomla\includes\framework.php (file 2)

**Comment by laoneo** [comment]
YOu can also test the second file when you go to the administrator subfolder and do the same. You can do it either from the joomla root or the administrator subfolder.

**Comment by HLeithner** [comment]
thanks

**Pull request merged**